src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImage.groovy

```diff
@@ -117,4 +122,38 @@ class DockerBuildImage extends AbstractDockerRemoteApiTask implements RegistryCr
         imageId = response.awaitImageId()
         logger.quiet "Created image with ID '$imageId'."
     }
-}
\ No newline at end of file
+
+    void saveImageId() {
+        def state = new TaskStateHelper(name, project)
+        state.put("imageId", getImageId())
+    }
+
+    boolean previouslyBuiltImageExists() {
+        def imageExists = false
+
+        def prevImageId = new TaskStateHelper(name, project).get("imageId")
+        if (!prevImageId?.trim()) {
+            logger.info "No previously saved imageId exists"
+            return false
+        }
+
+        runInDockerClassPath { dockerClient ->
+            try {
+                dockerClient.inspectImageCmd(prevImageId).exec()
+                logger.info "Image ${prevImageId} found via call to inspectImage"
+
+                imageExists = true
+            } catch(Exception e) {
+                logger.info "Image ${prevImageId} not found via call to inspectImage"
+                imageExists = false
+            }
+        }
+
+        if (imageExists) {
```

Comment:
I think we can just move this logic into the `try` clause and return from there.